sensor-validate.c

```diff
@@ -1,28 +1,44 @@
+#include <cstddef>
 #include "sensor-validate.h"
 
-int _give_me_a_good_name(double value, double nextValue, double maxDelta) {
-  if(nextValue - value > maxDelta) {
-    return 0;
+int checkChargeFluctuations(double value, double nextValue, double maxDelta) 
+{
+  int retFluctuationStatus = NO_DEVIATION_DETECTED;
+  if((nextValue - value) > maxDelta) 
+  {
+    retFluctuationStatus = READINGS_DEVIATION_DETECTED;
   }
-  return 1;
+  return retFluctuationStatus;
 }
 
-int validateSOCreadings(double* values, int numOfValues) {
+int validateChargeFluctuationStatus(double* inputParam, int numOfValues, int inputChargeType) 
+{
+  int retFluctuationStatus = NO_DEVIATION_DETECTED;
+  double retMaxThreshold = 0.0;
   int lastButOneIndex = numOfValues - 1;
-  for(int i = 0; i < lastButOneIndex; i++) {
-    if(!_give_me_a_good_name(values[i], values[i + 1], 0.05)) {
-      return 0;
-    }
+  if(NULL == inputParam)
+  {
+	  retFluctuationStatus = INVALID_PARAMETER_REQUESTED;
+	  return retFluctuationStatus;
   }
-  return 1;
+  retMaxThreshold = identifyChargeTypeForValidation(inputChargeType);
+  for(int i = 0; i < lastButOneIndex; i++) 
+  {
+    retFluctuationStatus &= checkChargeFluctuations(inputParam[i], inputParam[i + 1], retMaxThreshold);
```

Comment:
Good idea to make enum instead of 1 and 0. However, the use of enum with &= is not conventional. Because more enums can get added later, and this logic may be forgotten when we add.
Suggest to call out the aggregation of results in a separate function